src/Storage/Controllers/InstancesController.cs

```diff
@@ -359,35 +359,81 @@ public async Task<ActionResult<Instance>> Post(string appId, [FromBody] Instance
                 return Forbid();
             }
 
-            Instance storedInstance = new Instance();
+            Instance instanceToCreate, storedInstance;
             try
             {
                 DateTime creationTime = DateTime.UtcNow;
                 string userId = GetUserId();
 
-                Instance instanceToCreate = CreateInstanceFromTemplate(appInfo, instance, creationTime, userId);
+                instanceToCreate = CreateInstanceFromTemplate(appInfo, instance, creationTime, userId);
+            }
+            catch (Exception createFromTemplateEx)
+            {
+                _logger.LogError(createFromTemplateEx, "Unable to create {appId} instance from template for {instance.InstanceOwner.PartyId}, no cleanup necessary.", appId, instance.InstanceOwner.PartyId);
+                return StatusCode(500, $"Unable to create {appId} instance for {instance.InstanceOwner?.PartyId} due to {createFromTemplateEx.Message}");
```

Comment:
Inconsistent expectation of instance.InstanceOwner. First statement expect it to exist. The second does not.

Reply:
Now that we're looking at the code. I think it is best to not include exception message in the response.